File: client/dialog.lua
UI.DialogResp = {}
UI.Dialog = function(id, data, cb)
	if type(id) ~= "string" then
		error("UI: Missing required id")
		return false
	end
	if type(data.type) ~= "string" or type(data.title) ~= "string" or type(data.content) ~= "string" or type(data.posBtn) ~= "string" or type(data.negBtn) ~= "string" or type(data.posMsg) ~= "string" or type(data.negMsg) ~= "string" then
		error("UI: Missing required parameters for notify")
		return false
	end
	if type(cb) ~= "function" then
		error("UI: Missing required callback function")
		return false
	end

	UI.DialogResp[id] = cb

	SendNUIMessage({
		action = 'dialog',
		id = id,
		type = data.type,
		title = data.title,
		content = data.content,
		posBtn = data.posBtn,
		negBtn = data.negBtn,
		posMsg = data.posMsg,
		negMsg = data.negMsg
	})
	return true
end

RegisterNUICallback('atl_dialog_resp', function(data, cb)
	if UI.DialogResp[data.id] then
		UI.DialogResp[data.id](data.resp)
	end
	UI.DialogResp[data.id] = nil
	cb({})
end)

File: web/src/common/dialog.js
const TYPES = {
	success: 'success',
	info: 'info',
	warning: 'warning',
	error: 'error',
}

export const Dialog = (type, title, content, posBtn, negBtn, posMsg, negMsg) => {
	if (window.$dialog) {
		const t = TYPES[type.toLowerCase()]
		if (t) {
			window.$dialog[t]({
				title: title,
				content: content,
				positiveText: posBtn,
				negativeText: negBtn,
				maskClosable: false,
				closable: false,
				closeOnEsc: false,
				onPositiveClick: () => {
					window.$notify['success'](posMsg)
				},
				onNegativeClick: () => {
					window.$notify['error'](negMsg)
				}
			})
			return
		}
	}
	console.log('Could not send notification')
}